include/envoy/upstream/outlier_detection.h

```diff
@@ -20,6 +20,17 @@ typedef std::shared_ptr<const HostDescription> HostDescriptionConstSharedPtr;
 
 namespace Outlier {
 
+/**
+ * Non-HTTP result of requests/operations.
+ */
+enum class Result {
+  SUCCESS,        // Successfully established a connection or completed a request.
+  TIMEOUT,        // Timed out while connecting or executing a request.
+  CONNECT_FAILED, // Remote host rejected the connection.
+  REQUEST_FAILED, // Request was not completed successfully.
```

Comment:
Maybe clarify in comments that this only makes sense where the notion of request can be understood by Envoy on a connection. It might be useful to understand what `Result` fields make sense when this is a dumb TCP proxy vs. a more intelligent Redis/Mongo etc. proxy.